File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int Sum = 0, maxSum = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length; i++) {
            Sum += nums[i];
            maxSum = Math.max(maxSum, Sum);
            if (Sum < 0) Sum = 0;
        }
        return maxSum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        Sum, maxSum = 0, -10000
        for i in range(len(nums)): 
            Sum += nums[i]
            maxSum = max(maxSum, Sum)
            if Sum < 0: Sum = 0
        return maxSum
